--- a/src/modules/hyprland/workspace.cpp
+++ b/src/modules/hyprland/workspace.cpp
@@ -71,15 +71,15 @@ bool Workspace::handleClicked(GdkEventButton* bt) const {
     try {
       if (id() > 0) {  // normal
         if (m_workspaceManager.moveToMonitor()) {
-          m_ipc.getSocket1Reply("dispatch focusworkspaceoncurrentmonitor " + std::to_string(id()));
+          m_ipc.getSocket1Reply("dispatch hl.dsp.focus({workspace=" + std::to_string(id()) + "})");
         } else {
-          m_ipc.getSocket1Reply("dispatch workspace " + std::to_string(id()));
+          m_ipc.getSocket1Reply("dispatch hl.dsp.focus({workspace=" + std::to_string(id()) + "})");
         }
       } else if (!isSpecial()) {  // named (this includes persistent)
         if (m_workspaceManager.moveToMonitor()) {
-          m_ipc.getSocket1Reply("dispatch focusworkspaceoncurrentmonitor name:" + name());
+          m_ipc.getSocket1Reply("dispatch hl.dsp.focus({workspace=\"" + name() + "\"})");
         } else {
-          m_ipc.getSocket1Reply("dispatch workspace name:" + name());
+          m_ipc.getSocket1Reply("dispatch hl.dsp.focus({workspace=\"" + name() + "\"})");
         }
       } else if (id() != -99) {  // named special
